posthog/models/insight.py

```diff
@@ -174,12 +181,19 @@ class Meta:
 def generate_insight_cache_key(insight: Insight, dashboard: Optional[Dashboard]) -> str:
     try:
         if insight.query is not None:
-            # TODO: dashboard filtering needs to know how to override queries and date ranges 😱
-            q = insight.query
+            dashboard_filters = dashboard.filters if dashboard else None
+
+            if dashboard_filters:
+                from posthog.hogql_queries.apply_dashboard_filters import apply_dashboard_filters
+
+                q = apply_dashboard_filters(insight.query, dashboard_filters, insight.team)
+            else:
+                q = insight.query
+
             if q.get("source"):
                 q = q["source"]
 
-            return generate_cache_key("{}_{}".format(q, insight.team_id))
+            return generate_cache_key("{}_{}_{}".format(q, dashboard_filters, insight.team_id))
```

Comment:
1) Is there anything that also needs to be updated in `_cache_key` in query_runner? Probably not, as it's just using the generated SQL as its input...

2) Is this actually inlining a serialized version of the dashboard filters object, or a string that just says the class's name?

Reply:
2: From running this in a debugger, the string looks like `{'explain': None, 'filters': {'dateRange': {'date_from': '-4d', 'date_to': '-3d'}, 'properties': None}, 'kind': 'HogQLQuery', 'modifiers': None, 'query': 'select * from events where {filters}', 'response': None, 'values': None}_{'date_from': '-4d', 'date_to': '-3d'}_97`